torch/custom_class.h

```diff
@@ -177,6 +177,80 @@ class class_ {
     return *this;
   }
 
+  // Due to nullptr_t infer problem, we mark the default type of SetterFunc
+  // as GetterFunc and we check to make sure the set function is given.
+  template <typename GetterFunc, typename SetterFunc = GetterFunc>
+  class_& def_property(
+      std::string name,
+      GetterFunc getter_func,
+      SetterFunc setter_func = nullptr,
```

Comment:
We should probably have some check that makes sure `getter_func` and `setter_func` have reasonable schemas (e.g. getter must return something, setter cannot return anything).